sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -290,6 +301,30 @@ class wi_element<sycl::ext::oneapi::bfloat16, NumRows, NumCols, Use, Layout,
 #endif // __SYCL_DEVICE_ONLY__
 };
 
+template <typename T, size_t NumRows, size_t NumCols, use Use, layout Layout,
+          typename Group>
+inline __SYCL_ALWAYS_INLINE std::tuple<uint32_t, uint32_t>
+get_coord(wi_element<T, NumRows, NumCols, Use, Layout, Group> &we) {
+#ifdef __SYCL_DEVICE_ONLY__
+  __ocl_vec_t<uint32_t, 2> coord =
+      __spirv_JointMatrixGetElementCoordINTEL(we.M.spvm, we.idx);
+  const uint32_t row = coord[0];
+  const uint32_t col = coord[1];
+  return std::make_tuple(row, col);
+#else
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+}
+
+// To make host compilation possible, here the argument is not a wi_element
+// type, but just base data types e.g. float, int8 etc.
+template <typename T>
+inline __SYCL_ALWAYS_INLINE std::tuple<uint32_t, uint32_t> get_coord(T &we) {
+  throw runtime_error("joint matrix is not supported on host device.",
```

Comment:
This is fine, although FYI I don't think this runtime_error line can ever be reached at runtime, since an earlier runtime error will be thrown when the user tries the instantiate a joint_matrix on host.

Fine to leave this here though IMO. Maybe it is better to have it for code comprehension I guess.

Reply:
Agreed.